### DISCUSSION TITLE
Custom serial port settings

### DISCUSSION THREAD

**Discussion by LaQuiete1988** [aldas/modbus-tcp-client#163] [Q&A] [answered]
Hi!
Can't find a way to use Serial port with custom settings. 
Is there a way to replace $sttyModes array with another one?

**Comment by aldas** [comment, marked as answer]
just edit it in https://github.com/aldas/modbus-tcp-client/blob/master/examples/rtu_usb_to_serial.php

or if you are using https://github.com/aldas/modbus-tcp-client/blob/master/examples/rtu_usb_to_serial_stream.php

then pass it from constructor and you can change defaults
https://github.com/aldas/modbus-tcp-client/blob/b18323bf5bb2abb531d97d6f385bafb3a6c0c658/src/Network/SerialStreamCreator.php#L54


like that
```php
$connection = BinaryStreamConnection::getBuilder()
    ->setUri('/dev/ttyUSB0')
    ->setCreateStreamCallback(function (BinaryStreamConnection $conn) {
        $streamCreator = new SerialStreamCreator(['sttyModes' => SerialStreamCreator::DEFAULT_STTY_MODES]);
        return $streamCreator->createStream($conn);
    })
    ->setIsCompleteCallback(static function ($binaryData, $streamIndex): bool {
        return Packet::isCompleteLengthRTU($binaryData);
    })
    // delay this is crucial for some serial devices and delay needs to be long as 100ms (depending on the quantity)
    // or you will experience read errors ("stream_select interrupted") or invalid CRCs
    ->setDelayRead(100_000) // 100 milliseconds, serial devices may need delay between sending and received
    ->setLogger(new EchoLogger())
    ->build();
```

NB: I did not test it at the moment

**Reply by LaQuiete1988** [reply]
Cool! Thanks!
I use stream and this setCreateStreamCallback works perfectly!
